sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -1063,8 +1065,8 @@ ProgramManager::getDeviceImage(const std::string &KernelName,
                                     Context, Device);
       assert(Img && "No binary image found for kernel id");
     } else {
-      Img = getBinImageFromMultiMap(m_ServiceKernels, KernelName, Context,
-                                    Device);
+      Img = getBinImageFromMultiMap(m_ServiceKernels, std::string(KernelName),
```

Comment:
This string creation seems unnecessary. Can we work with the passed pointer without it, either directly or via `std::string_view`?

Reply:
Pass char* as a key to unordered_map for look up causes failure that's why I kept string here. Should i look into pass char* instead?